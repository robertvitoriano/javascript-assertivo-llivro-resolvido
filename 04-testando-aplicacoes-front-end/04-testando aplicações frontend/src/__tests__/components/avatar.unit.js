import { renderWithTheme } from "../../testUtils";
import Avatar from "../../components/avatar";
import { screen } from "@testing-library/react";
import theme from './../../styles/theme';

describe("<Avatar/>", () => {
  test.each([
    ['', false],
    ['http://someimage.com', false],
    ['', true],
    ['http://someimage.com', true],
  ])(
    'Should render the avatar correctly with %s src and minimal=%s',
    (src, minimal) => {
      const props = {
        src,
        minimal
      };

      renderWithTheme(<Avatar {...props} />);

      const avatar = screen.getByRole('figure');
      const avatarMissingIcon = screen.queryByTestId('avatar-missing-icon');

      if (avatarMissingIcon) {
        expect(avatar).toBeVisible();
        expect(avatarMissingIcon).toBeVisible();
      }

      if (minimal) {
        expect(avatar).toHaveStyle(`
          overflow: hidden;
          width: ${theme.spacing.double};
          height: ${theme.spacing.double};
          border-radius: ${theme.radius.rounded};
        `);
      }

      if (src) {
        const image = screen.getByRole('img');
        expect(image).toBeVisible();
      } else {
        expect(screen.queryByRole('img')).not.toBeInTheDocument();
      }
    }
  );
});
