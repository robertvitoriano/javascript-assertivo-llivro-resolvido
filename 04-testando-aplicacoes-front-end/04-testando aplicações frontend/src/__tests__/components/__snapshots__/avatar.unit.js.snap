// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> Should be able to render avatar component with minimalistic styles once minimal is true with a image 1`] = `
<figure
  class="sc-hKgILt fSRGMS"
>
  <img
    alt="avatar"
    src="http://someimage.com"
  />
</figure>
`;
